### DISCUSSION TITLE
Support for Golang application

### DISCUSSION THREAD

**Discussion by hariohmprasath** [kool-dev/kool#400] [Ideas]
Describe the feature

A clear and concise description of what the feature idea is. Please make sure you didn't find an existing issue with the same subject.

Go is quite popular for building server and command line based application. The whole build and packing can be done in a build container so the developer can just concentrate on write code, instead of worrying about the build & deployment platform. Pretty much all the top companies use Go as a defacto standard to build application that can run across multiple platforms

I don't see a repository for `docker-go` will be happy to contribute this, let me know if the team is interested in adding this support to platform

Backward compatibility concerns

Would this be BC?
No

Use cases

Golang applications

Examples

Server & command line applications built using Golang

**Comment by fabriciojs** [comment]
Helllo @hariohmprasath ! This is great, thanks for getting in touch!

About Golang we have two different matters, as you mentioned yourself: CLI tools and server applications.

As long as CLI tool goes, `kool` itself is a Golang CLI tool which is already developed with `kool` 😄 

We have created a simple, not-yet-advertised [preset for Go CLI tools](https://github.com/kool-dev/kool/blob/master/presets/golang-cli/kool.yml). It's is inspired by how we handle [`kool`](https://github.com/kool-dev/kool/blob/master/kool.yml) development itself. So, we have a good starting point for this.

For server applications, apart from some experiments with a preset for Goxygen, there hasn't been any more work. Certainly would be great to have it.

If you are willing to contribute, it will be most appreciated! I can see a couple of paths we can take to start:

- About the image: for the CLI we have been using official images since it's mostly for building only. Didn't see the need for a custom image, but implementing the preset for server apps might bring other needs that would justify having a custom/specialized image. We need to assess that.
- About server applications - we can have a generic preset, with options to support services like relational databases/cache/NoSQL databases.
- Still about server applications - we can have specific presets, i.e for [goxygen](https://github.com/Shpota/goxygen) or others.
